utils.h

```diff
@@ -0,0 +1,36 @@
+#ifndef _UTILS_H
```

Comment:
Maybe name the file as log_utils or log_file_utils for clearance?
And for OPEN_LOGFILE and CLOSE_LOGFILE, instead of each pair in mysqlna_azure::connect. The function pair RINIT/RSHUNDOWN or GINIT/GSHUTDOWN/MSHUTDOWN should be a better place. Since we can make sure it will be called, and if defined in RINT/RSHUTDOWN, it will be opened/closed only once per srcipt run, if defined in GINIT/GSHUTDOWN/MSHUTDOWN, it will be opened/closed only once per thread/process depending on PHP thread mode.